tests/profiles/googlefonts_test.py

```diff
@@ -3352,6 +3352,16 @@ def test_check_repo_zip_files():
             os.remove(filepath)
 
 
+def test_check_vertical_metrics():
+    check = CheckTester(googlefonts_profile,
+                        "com.google.fonts/check/vertical_metrics")
+
+    ttFont = TTFont(TEST_FILE("akshar/Akshar[wght].ttf"))
+    assert_results_contain(check(ttFont),
+                           FAIL, 'bad-hhea.lineGap',
+                           'hhea.lineGap is "150" it should be 0')
+
+
```

Comment:
Please elaborate

Reply:
you're only checking that the hhea linegaps are set to 0. You should also write another test to check the absolute sum of the hhea metrics.

Reply:
That's good and kind of was I was missing here, but it involves checking the supported scripts of the font as we require 1200 upm sum only for Latin/Greek/Cyrillic per out specs. Is there an easy interface in place in fontbakery?

Reply:
afaik we don't have a method to determine this. However, since this check is only raising a warn, it would be nice to see this check trigger regardless of what scripts are in the font. Spec also say

Reply:
Oh, now I see what you mean. You want me to include this in the tests, not add another check for range not being 1200. I'm slow sometimes.

Reply:
yes :-)

Reply:
I need help please. I added the WARN test. Now I get this message in `pytest`.
The check is not emitting a bare string. I have no idea what this is about.


```
tests/profiles/googlefonts_test.py F....                                                                                                                                                                           [100%]

======================================================================================================== FAILURES ========================================================================================================
______________________________________________________________________________________________ test_check_vertical_metrics _______________________________________________________________________________________________

    def test_check_vertical_metrics():
        check = CheckTester(googlefonts_profile,
                            "com.google.fonts/check/vertical_metrics")
    
        ttFont = TTFont(TEST_FILE("akshar/Akshar[wght].ttf"))
        assert_results_contain(check(ttFont),
                               FAIL, 'bad-hhea.lineGap',
                               'hhea.lineGap is "150" it should be 0')
    
        ttFont['hhea'].lineGap = float('inf')
>       assert_results_contain(check(ttFont),
                               WARN, 'bad-hhea-range',
                               'if font hhea are greater than 1.5 * upm')

tests/profiles/googlefonts_test.py:3365: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

check_results = [(<Status FAIL>, OS/2.sTypoLineGap is "150" it should be 0 [code: bad-OS/2.sTypoLineGap]), (<Status FAIL>, hhea.lineGa...f" it should be 0 [code: bad-hhea.lineGap]), (<Status ERROR>, FailedCheckError("Failed with KeyError: 'hhea.ascent'"))]
expected_status = <Status WARN>, expected_msgcode = 'bad-hhea-range', reason = 'if font hhea are greater than 1.5 * upm', ignore_error = None

    def assert_results_contain(check_results,
                               expected_status,
                               expected_msgcode,
                               reason=None,
                               ignore_error=None):
        """
        This helper function is useful when we want to make sure that
        a certain log message is emited by a check but it can be in any
        order among other log messages.
        """
        from fontbakery.message import Message
        from fontbakery.checkrunner import PASS, DEBUG, ERROR
    
        if not isinstance(expected_msgcode, str):
            raise Exception(
                "The expected message code must be a string")
    
        if not reason:
            reason = f"[{expected_msgcode}]"
    
        print(f"Test {expected_status} {reason}")
        check_results = list(check_results)
    
        for status, _ in check_results:
            if ignore_error and status == ERROR:
                print(ignore_error)
                return None
    
        for status, msg in check_results:
            if status not in [PASS, DEBUG] and not isinstance(msg, Message):
>               raise Exception(
                    f"Bare Python strings are no longer supported in result values.\n"
                    f"Please use the Message class to wrap strings and to give them"
                    f" a keyword useful for identifying them (in bug reports as well as"
                    f" in the implementation of reliable unit tests).\n"
                    f"(Bare string: {msg!r})")
E               Exception: Bare Python strings are no longer supported in result values.
E               Please use the Message class to wrap strings and to give them a keyword useful for identifying them (in bug reports as well as in the implementation of reliable unit tests).
E               (Bare string: FailedCheckError("Failed with KeyError: 'hhea.ascent'"))

/usr/local/lib/python3.9/site-packages/fontbakery/codetesting.py:219: Exception
-------------------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------------------
Test <Status FAIL> hhea.lineGap is "150" it should be 0
Test <Status WARN> if font hhea are greater than 1.5 * upm
================================================================================================ short test summary info =================================================================================================
FAILED tests/profiles/googlefonts_test.py::test_check_vertical_metrics - Exception: Bare Python strings are no longer supported in result values.
```

Reply:
`<Status ERROR>, FailedCheckError("Failed with KeyError: 'hhea.ascent'")`